invenio_mail/ext.py

```diff
@@ -67,5 +67,11 @@ def init_config(self, app):
     def print_email(self, message, app):
         """Print mail to stream."""
         with self._lock:
-            self.stream.write("{0}\n{1}\n".format(message.as_string(), '-'*79))
-            self.stream.flush()
+            invenio_mail = app.extensions['invenio-mail']
```

Comment:
This is actually bug. I believe the print_email should be taken out of the extension - i.e. it should be something like:

```python
class InvenioMail(object):
    """Invenio-Mail extension.
    Mails are only printed to the stream if ``MAIL_SUPPRESS_SEND`` is ``True``.
    :param app: Flask application object.
    :param stream: Stream to print emails to. Defaults to ``sys.stdout``.
    """

    def __init__(self, app=None, stream=None):
        """Extension initialization."""
        self.stream = stream or sys.stdout
        if app:
            self.init_app(app)

    def init_app(self, app):
        """Flask application initialization."""
        self.init_config(app)
        if 'mail' not in app.extensions:
            Mail(app)
        if app.config.get('MAIL_SUPPRESS_SEND', False) or app.debug:
            email_dispatched.connect(print_email)
        app.extensions['invenio-mail'] = self

    def init_config(self, app):
        """Initialize configuration."""
        app.config.setdefault('MAIL_DEBUG', app.debug)
        app.config.setdefault('MAIL_SUPPRESS_SEND', app.debug or app.testing)


def print_email(message, app):
    """Print mail to stream."""
    stream = app.extensions['invenio-mail'].stream
    stream.write("{0}\n{1}\n".format(message.as_string(), '-'*79))
    stream.flush()
```